x-pack/plugin/ent-search/qa/rest/src/yamlRestTest/resources/rest-api-spec/test/entsearch/340_connector_update_status.yml

```diff
@@ -0,0 +1,92 @@
+setup:
+  - skip:
+      version: " - 8.12.99"
+      reason: Introduced in 8.13.0
+
+  - do:
+      connector.put:
+        connector_id: test-connector
+        body:
+          index_name: search-1-test
+          name: my-connector
+          language: pl
+          is_native: false
+          service_type: super-connector
+
+---
+"Update Connector Status":
+  - do:
+      connector.update_status:
+        connector_id: test-connector
+        body:
+          status: needs_configuration
+
+  - match: { result: updated }
+
+  - do:
+      connector.get:
+        connector_id: test-connector
+
+  - match: { status: needs_configuration }
+
+  - do:
+      connector.update_status:
+        connector_id: test-connector
+        body:
+          status: configured
+
+  - match: { result: updated }
+
+  - do:
+      connector.get:
+        connector_id: test-connector
+
+  - match: { status: configured }
+
+---
+"Update Connector Status - 404 when connector doesn't exist":
+  - do:
+      catch: "missing"
+      connector.update_status:
+        connector_id: test-non-existent-connector
+        body:
+          status: needs_configuration
+
+---
+"Update Connector Status - 400 status code when connector_id is empty":
+  - do:
+      catch: "bad_request"
+      connector.update_status:
+        connector_id: ""
+        body:
+          status: needs_configuration
+
+---
+"Update Connector Status - 400 status code when payload is not string":
+  - do:
+      catch: "bad_request"
+      connector.update_status:
+        connector_id: test-connector
+        body:
+          status:
+            field_1: test
+            field_2: something
+
+---
+"Update Connector Status - 400 status code when invalid status":
+  - do:
+      catch: "bad_request"
+      connector.update_status:
+        connector_id: test-connector
+        body:
+          status: _invalid-status
+
+
+---
+"Update Connector Status - 400 status code when invalid status transition":
+  - do:
+      catch: "bad_request"
+      connector.update_status:
+        connector_id: test-connector
+        body:
+          status: connected # created -> connected transition is invalid
```

Comment:
Nit: should we include a check that the expected error is output?

Reply:
Good point, I will look into it :)